ozaria/site/components/interactive/PageInteractiveEditor/index.vue

```diff
@@ -0,0 +1,208 @@
+<script>
+import { getInteractive, putInteractive, postInteractive, getAllInteractives } from '../../../api/interactive'
+import Interactive from '../../../models/Interactive'
+import ListItem from '../../common/BaseListItem'
+import Ajv from 'ajv'
+import ozariaUtils from 'app/core/ozariaUtils'
+
+require('lib/setupTreema')
+
+module.exports = Vue.extend({
+  props: {
+    slug: String
+  },
+  data: () => ({
+    interactive: null,
+    treema: null,
+    interactiveList: null,
+    valid: true,
+    state: {
+      saving: false
+    },
+    interactiveSlug: ""
+  }),
+  components: {
+    'list-item': ListItem
+  },
+  async created () {
+    if (!me.hasInteractiveAccess()) {
+      alert('You must be logged in as an admin to use this page.')
+      return application.router.navigate('/editor', { trigger: true })
+    }
+    console.log(`Got the slug: ${this.slug}`)
+    this.interactiveSlug = this.slug
+    if (this.interactiveSlug)  {
+      await this.fetchInteractive(this.interactiveSlug)
+    }
+    else {
+      await this.fetchList()
+    }
+  },
+  methods: {
+    /**
+     * Fetch and populate treema with interactive slug.
+     * Clears the list
+     */
+    async fetchInteractive(slug) {
+      this.interactiveList = null
+      this.interactiveSlug = slug
+      try {
+        this.interactive = new Interactive(await getInteractive(slug))
+      } catch (e) {
+        noty({ text: `Error finding slug '${slug}'.`, type: 'error', timeout: '1000' })
+        return this.fetchList()
+      }
+      const data = $.extend(true, {}, this.interactive.attributes)
+      const el = $(`<div></div>`);
+      const treemaOptions = {
+        data: data,
+        schema: Interactive.schema,
+        callbacks: {
+          change: this.onTreemaChanged
+        }
+      }
+      const treema = this.treema = TreemaNode.make(el, treemaOptions)
+      treema.build()
+      $(this.$refs.treemaEditor).append(el)
```

Comment:
Did this start working. What had prevented it from working in the template? 😄

Reply:
Haha, yeah I played around it for a while and it started working. 😄 
I don't know exactly what made it work, but I think changing the variable that I was using in the template for `v-if-else` made things work. There was something wrong with the variable mutation earlier.